frame/authority-discovery/src/lib.rs

```diff
@@ -74,15 +82,15 @@ impl<T: Config> pallet_session::OneSessionHandler<T::AccountId> for Module<T> {
 		Self::initialize_keys(&authorities.map(|x| x.1).collect::<Vec<_>>());
 	}
 
-	fn on_new_session<'a, I: 'a>(changed: bool, validators: I, queued_validators: I)
+	fn on_new_session<'a, I: 'a>(_changed: bool, validators: I, queued_validators: I)
 	where
 		I: Iterator<Item = (&'a T::AccountId, Self::Key)>,
 	{
 		// Remember who the authorities are for the new and next session.
-		if changed {
-			let keys = validators.chain(queued_validators).map(|x| x.1).collect::<BTreeSet<_>>();
-			Keys::put(keys.into_iter().collect::<Vec<_>>());
-		}
+		let keys = validators.map(|x| x.1).collect::<Vec<_>>();
```

Comment:
Why is the check on changed removed?

Reply:
Does it apply to the current session only? I.e. do we have to store `NextKeys` irregardless?